File: buyandsellstock.cpp
class Solution {
public:

       int maxProfit(vector<int>&prices){

           int n = prices.size();

           int mini = prices[0];
           int maxi = prices[1];
          
           int ans = 0;

           for(int i=0;i<n;i++){
               if(price[i] < mini){
                  mini = price[i];
                  maxi = price[i];

               }

               maxi = max(maxi, price[i]);
               ans = max(ans, maxi-mini);
           }

           return ans;
       }

};

File: buyandsellstock.py
class Solution:
    def maxProfit(self, prices):
        n = len(prices)
        
        mini = prices[0]
        maxi = prices[1]
        ans = 0
        
        for i in range(n):
            if prices[i] < mini:
                mini = prices[i]
                maxi = prices[i]
            
            maxi = max(maxi, prices[i])
            ans = max(ans, maxi - mini)
        
        return ans
